src/crsAdapter/SchmetterlingNeoAdapter.js

```diff
@@ -0,0 +1,345 @@
+        crsData.converted.remark = crsData.normalized.remark;
+
+        this.assignServices(crsData);
+        this.assignTravellers(crsData);
+
+        crsData.build = crsData.converted;
+
+        return crsData;
+    }
+
+    assignServices(crsData) {
+        crsData.normalized.services.forEach((service, index) => {
+            const crsServiceObject = {};
+
+            crsServiceObject.marker = service.marker;
+            crsServiceObject.serviceType = service.type;
+            crsServiceObject.serviceCode = service.code;
+            crsServiceObject.accommodation = service.accommodation;
+            crsServiceObject.fromDate = service.fromDate;
+            crsServiceObject.toDate = service.toDate;
+            crsServiceObject.occupancy = service.occupancy;
+            crsServiceObject.quantity = service.quantity;
+            crsServiceObject.travellerAssociation = service.travellerAssociation;
+
+            crsData.converted.services[index] = crsServiceObject;
+        });
+    }
+
+    assignTravellers(crsData) {
+        crsData.normalized.travellers.forEach((traveller, index) => {
+            const crsTravellerObject = {};
+
+            crsTravellerObject.title = traveller.title;
+            crsTravellerObject.name = traveller.name;
+            crsTravellerObject.discount = traveller.age;
+
+            crsData.converted.travellers[index] = crsTravellerObject;
+        });
+    }
+
+    sendData(crsData = {}) {
+        return this.sendCrsObject(crsData.build).then(() => {
+            return this.cancel();
+        });
+    }
+
+    cancel() {
+        return new Promise((resolve) => {
+            try {
+                this.getConnection().requestService(
+                    'popups.close',
+                    null,
+                    this.createCallbackObject(resolve, null, 'cancel error')
+                );
+            } catch (error) {
+                this.logger.error(error);
+                throw new Error('connection::popups.close: ' + error.message);
+            }
+        });
+    }
+
+    /**
+     * @private
+     * @param options object
+     * @returns {Promise}
+     */
+    createConnection(options = {}) {
+        return new Promise((resolve) => {
+            const connectToCRS = () => {
+                let callbackObject = this.createCallbackObject(
+                    resolve,
+                    () => {
+                        this.logger.info('connected to Schmetterling NEO');
+                        this.connection = window.catalog;
+                    },
+                    'connection not possible'
+                );
+
+                callbackObject.scope = window;
+
+                window.catalog.dest = connectionUrl;
+                window.catalog.connect(callbackObject);
+            };
+
+            const cleanUrl = (url = '') => {
+                if (!url) return;
+
+                return 'https://' + url.replace("https://", '').split('/')[0];
```

Comment:
Does this only works secured with http"s"?

Reply:
Yes, ... hopefully. Else our whole application will not work within the CRS because it is served via https.